mypy/dmypy_server.py

```diff
@@ -274,16 +273,17 @@ def initialize_fine_grained(self, sources: List[mypy.build.BuildSource]) -> Dict
         self.fine_grained_manager = mypy.server.update.FineGrainedBuildManager(manager, graph)
         self.fine_grained_initialized = True
         self.previous_sources = sources
-        self.fscache.flush()
 
-        # If we are using the fine-grained cache, build hasn't actually done
-        # the typechecking on the updated files yet.
+        # If we are using the fine-grained cache, build might not have
+        # actually done the typechecking on the updated files yet.
         # Run a fine-grained update starting from the cached data
         if self.options.use_fine_grained_cache:
             # Pull times and hashes out of the saved_cache and stick them into
             # the fswatcher, so we pick up the changes.
             for state in self.fine_grained_manager.graph.values():
                 meta = state.meta
+                # If there isn't a meta, that means the current
+                # version got checked in the initial build.
```

Comment:
Relying on this logic here seems a bit fragile. Maybe move the `meta is None` check to a helper method in `State` and use the helper method here instead?

Reply:
Changing it to use `is_cache_skeleton` instead.